### PR TITLE
[macro] Literal expansion

### PR DESCRIPTION
**Pull request by pkriens**
The ${range} macro created an error when it was used without the @ target set. Added a feature to indicate to the macro processor that the replacement should be the literal text of the macro so it can be expanded again later.

Signed-off-by: Peter Kriens <peter.kriens@aqute.biz>

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java [hunk 1]
```diff
@@ -933,6 +952,10 @@ public String _range(String args[]) {
 		return sb.toString();
 	}
 
+	boolean isLocalTarget(String string) {
+		return string.matches("\\$(\\{@\\}|\\[@\\]|\\(@\\)|<@>|«@»|‹@›)");
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java]: Does this also need to handle `$@`?

File: biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java [hunk 2]
```diff
@@ -899,12 +912,18 @@ String version(Version version, String mask) {
...
+			if (isLocalTarget(string))
+				return LITERALVALUE;
+
+			version = new Version(string);
+		}
 		else {
 			String v = domain.getProperty("@");
-			if (v == null)
-				return null;
+			if (v == null || v.contains("@"))
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java]: I think you need to remove the `contains("@")` test

**Comment by bjhargrave** [comment]
Your commit comment should also note that this fixes https://github.com/bndtools/bnd/issues/1350.

File: biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java (hunk 1 above)
Comment (pkriens) [inline biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java]: As far as I can see $@ is illegal. You always need to encloses it.

**Pull request merged**